File: OrderAnalysis-ordergoods-interface/src/main/java/com/orderanalsis/ordergoods/service/CustAmtRankService.java
package com.orderanalsis.ordergoods.service;
import java.util.List;
import com.orderanalsis.pojo.TbCustAmtRank;

import entity.PageResult;
/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface CustAmtRankService {

	/**
	 * 返回全部列表
	 * @return
	 */
	public List<TbCustAmtRank> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
	public PageResult findPage(int pageNum,int pageSize);
	
	
	/**
	 * 增加
	*/
	public void add(TbCustAmtRank custAmtRank);
	
	
	/**
	 * 修改
	 */
	public void update(TbCustAmtRank custAmtRank);
	

	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public TbCustAmtRank findOne(Long id);
	
	
	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(Long [] ids);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(TbCustAmtRank custAmtRank, int pageNum,int pageSize);
	
}

File: OrderAnalsis-ordergoods-web/src/main/webapp/js/service/custAmtRankService.js
//服务层
app.service('custAmtRankService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../custAmtRank/findAll.do');		
	}
	//分页 
	this.findPage=function(page,rows){
		return $http.get('../custAmtRank/findPage.do?page='+page+'&rows='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('../custAmtRank/findOne.do?id='+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('../custAmtRank/add.do',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('../custAmtRank/update.do',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('../custAmtRank/delete.do?ids='+ids);
	}
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('../custAmtRank/search.do?page='+page+"&rows="+rows, searchEntity);
	}    	
});
